sdcm/nemesis.py

```diff
@@ -4863,46 +4845,48 @@ def disrupt_add_remove_dc(self) -> None:
             raise UnsupportedNemesis(
                 "add_remove_dc skipped for multi-dc scenario (https://github.com/scylladb/scylla-cluster-tests/issues/5369)")
         InfoEvent(message='Starting New DC Nemesis').publish()
+
         node = self.cluster.data_nodes[0]
+        stress_tests = []
         system_keyspaces = ["system_distributed", "system_traces"]
-        if not node.raft.is_consistent_topology_changes_enabled:  # auth-v2 is used when consistent topology is enabled
-            system_keyspaces.insert(0, "system_auth")
-        self._switch_to_network_replication_strategy(self.cluster.get_test_keyspaces() + system_keyspaces)
+        if not node.raft.is_consistent_topology_changes_enabled:
+            system_keyspaces.append("system-auth")
+
         datacenters = list(self.tester.db_cluster.get_nodetool_status().keys())
         self.tester.create_keyspace("keyspace_new_dc", replication_factor={
                                     datacenters[0]: min(3, len(self.cluster.data_nodes))})
-        node_added = False
+
         with ExitStack() as context_manager:
             def finalizer(exc_type, *_):
                 # in case of test end/killed, leave the cleanup alone
                 if exc_type is not KillNemesis:
                     with self.cluster.cql_connection_patient(node) as session:
                         session.execute('DROP KEYSPACE IF EXISTS keyspace_new_dc')
-                    if node_added:
-                        self.cluster.decommission(new_node)
+                        if self.cluster.get_node_status_dictionary(new_node.ip_address):
+                            [st.kill() for st in stress_tests]
```

Comment:
This is where I had a problem. When removing dc while running the stress test I was getting stress test errors such as:
`sstable - Could not create SSTable component /var/lib/scylla/data/keyspace1/standard1-306a9070013511f0808e70056419ac5f/me-3gol_04hz_4igrk29bkilslik7ig-big-CRC.db. Found exception: std::filesystem::__cxx11::filesystem_error (error system:2, filesystem error: open failed: No such file or directory `
`compaction - [Scrub compaction keyspace1.standard1] unrecoverable error: /var/lib/scylla/data/keyspace1/standard1-306a9070013511f0808e70056419ac5f/me-3gol_04hz_4igrk29bkilslik7ig-big-CRC.db: file not found`

No similar problems occurs when stress tests are killed. Is this the correct approach?

Reply:
If so, what is better way how to do it? `ClusterTester.tester.kill_stress_thread()` or `DockerBasedStressThread.kill()`

Reply:
I think you are correct here. I am investigating this now.